src/webgpu/shader/validation/expression/call/builtin/insertBits.spec.ts

```diff
@@ -0,0 +1,212 @@
+const builtin = 'insertBits';
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf, objectsToRecord } from '../../../../../../common/util/data_tables.js';
+import { range } from '../../../../../../common/util/util.js';
+import {
+  Type,
+  kConcreteIntegerScalarsAndVectors,
+  kFloatScalarsAndVectors,
+  u32,
+} from '../../../../../util/conversion.js';
+import { linearRange } from '../../../../../util/math.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import {
+  fullRangeForType,
+  kConstantAndOverrideStages,
+  stageSupportsType,
+  validateConstOrOverrideBuiltinEval,
+} from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kValuesTypes = objectsToRecord(kConcreteIntegerScalarsAndVectors);
+
+g.test('values')
+  .desc(
+    `
+Validates that constant evaluation and override evaluation of ${builtin}() never errors on valid inputs
+`
+  )
+  .params(u =>
+    u
+      .combine('stage', kConstantAndOverrideStages)
+      .combine('type', keysOf(kValuesTypes))
+      .filter(u => stageSupportsType(u.stage, kValuesTypes[u.type]))
+      .beginSubcases()
+      .expand('value', u => fullRangeForType(kValuesTypes[u.type]))
+      .expand('newbits', u => fullRangeForType(kValuesTypes[u.type]))
+      .expand('offset', _ => linearRange(0, 32, 1).map(f => Math.floor(f)))
```

Comment:
Looking at `linearRange`, this is just going to return an array of `[0]` due to special casing of the last param === 1. I think, you possibly wanted `0, 32, 32` as the param to get 32 entries in the resulting array?

Reply:
Changed it a bit to prune the total number of cases a bit too. Otherwise the number of combinations was getting out of hand in terms of runtime.